src/api/DutchX.ts

```diff
@@ -0,0 +1,142 @@
+import { promisedContractsMap } from './contracts'
+import { promisedWeb3 } from './web3Provider'
+import { DutchExchange } from './types'
+import { TokenPair, Account } from 'types'
+
+export const promisedDutchX = init()
+
+const getCurrentAccount = async () => {
+  const web3 = await promisedWeb3
+
+  return web3.getCurrentAccount()
+}
+
+async function init(): Promise<DutchExchange> {
+  const contractsMap = await promisedContractsMap
+
+  const getExchange = ({ sell, buy }: TokenPair) => {
+    const exchange = contractsMap[`DutchExchange${sell}${buy}`]
+
+    if (!exchange) throw new Error(`No DutchExchange contract for ${sell}->${buy} token pair`)
+
+    return exchange
+  }
+
+  const fillDefaultIndexAndAccount = (pair: TokenPair, index: number, account: Account) => Promise.all([
+    index === undefined ? getAuctionIndex(pair) : index,
+    account === undefined ? getCurrentAccount() : account,
+  ])
+
+  const getAuctionIndex = async (pair: TokenPair) => {
+    const index = await getExchange(pair).auctionIndex()
+
+    return index.toNumber()
+  }
+
+  const getClosingPrice = async (pair: TokenPair, index?: number) => {
+    if (index === undefined) index = await getAuctionIndex(pair) - 1
+
+    const [num, den] = getExchange(pair).closingPrices(index)
+
+    return num.div(den).toString()
```

Comment:
Not super positive with the changes that will come - but I think we should be careful of this return of `num/den` as we may need to divide the inverse when calculating the other side of Token Exchanges _e.g `ETH/GNO` && `GNO/ETH`_